lib/pages/modlog_page.dart

```diff
@@ -0,0 +1,519 @@
+  final int communityId;
+
+  const ModlogPage.forInstance({
+    @required this.instanceHost,
+  })  : assert(instanceHost != null),
+        communityId = null,
+        name = instanceHost;
+
+  const ModlogPage.forCommunity({
+    @required this.instanceHost,
+    @required this.communityId,
+    @required String communityName,
+  })  : assert(instanceHost != null),
+        assert(communityId != null),
+        assert(communityName != null),
+        name = '!$communityName';
+
+  @override
+  Widget build(BuildContext context) {
+    final page = useState(1);
+    // will be set true when a fetch returns 0 results
+    final isDone = useState(false);
+
+    final modlogFut = useMemoized(
+      () => LemmyApiV2(instanceHost).run(
+        GetModlog(
+          communityId: communityId,
+          page: page.value,
+        ),
+      ),
+      [page.value],
+    );
+
+    return Scaffold(
+      appBar: AppBar(title: Text("$name's modlog")),
+      body: SingleChildScrollView(
+        child: Column(
+          children: [
+            FutureBuilder<Modlog>(
+              key: ValueKey(modlogFut),
+              future: modlogFut,
+              builder: (context, snapshot) {
+                if (!snapshot.hasData) {
+                  return const Center(child: CircularProgressIndicator());
+                } else if (snapshot.hasError) {
+                  return Center(
+                      child: Text('Error: ${snapshot.error?.toString()}'));
+                }
+                final modlog = snapshot.data;
+
+                if (modlog.added.length +
+                        modlog.addedToCommunity.length +
+                        modlog.banned.length +
+                        modlog.bannedFromCommunity.length +
+                        modlog.lockedPosts.length +
+                        modlog.removedComments.length +
+                        modlog.removedCommunities.length +
+                        modlog.removedPosts.length +
+                        modlog.stickiedPosts.length ==
+                    0) {
+                  WidgetsBinding.instance
+                      .addPostFrameCallback((_) => isDone.value = true);
+
+                  return const Center(child: Text('no more logs to show'));
+                }
+
+                return _ModlogTable(modlog: modlog);
+              },
+            ),
+            Padding(
+              padding: const EdgeInsets.all(8),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  TextButton(
+                    onPressed: page.value != 1 ? () => page.value-- : null,
+                    child: const Icon(Icons.skip_previous),
+                  ),
+                  TextButton(
+                    onPressed: isDone.value ? null : () => page.value++,
+                    child: const Icon(Icons.skip_next),
+                  ),
+                ],
+              ),
+            )
```

Comment:
add `BottomSafe` for those pesky iPhone users
![image](https://user-images.githubusercontent.com/10037914/108446562-a1160800-725e-11eb-96bf-727c15aa3009.png)

Reply:
It is in a Scaffold so a safearea was already applied no?

Reply:
No! We had this convo so many times